src/gsy_e/gsy_e_core/simulation.py

```diff
@@ -523,9 +543,55 @@ def _get_current_market_time_slot(self, slot_number: int) -> DateTime:
         return (self.area.config.start_date + (slot_number * self.area.config.slot_length)
                 if GlobalConfig.IS_CANARY_NETWORK else self.area.now)
 
+    def _execute_coefficient_simulation(
+            self, slot_resume: int, tick_resume: int, console: NonBlockingConsole = None) -> None:
+
+        slot_count, slot_resume, tick_resume = (
+            self._time.calculate_total_initial_ticks_slots(
+                self._setup.config, slot_resume, tick_resume, self.area))
+
+        for slot_no in range(slot_resume, slot_count):
+            self._handle_paused(console)
+
+            self.progress_info.update(slot_no, slot_count, self._time, self._setup.config)
+
+            self.area.cycle_coefficients_trading(self.progress_info.current_slot_time)
+
+            global_objects.profiles_handler.update_time_and_buffer_profiles(
+                self._get_current_market_time_slot(slot_no))
+
+            self._results.update_and_send_results(
+                self.current_state, self.progress_info, self.area, self._status.status)
+            self._external_events.update(self.area)
+
+            gc.collect()
+            process = psutil.Process(os.getpid())
+            mbs_used = process.memory_info().rss / 1000000.0
+            log.debug("Used %s MBs.", mbs_used)
+
+            self._time.handle_slowdown_and_realtime(0, self._setup.config)
+
+            if self._status.stopped:
+                log.error("Received stop command for configuration id %s and job id %s.",
+                          gsy_e.constants.CONFIGURATION_ID, self._simulation_id)
+                sleep(5)
+                self._simulation_finish_actions(slot_count)
+                return
+
+            self._results.update_csv_on_market_cycle(slot_no, self.area)
+            self._status.handle_incremental_mode()
+
+        self._simulation_finish_actions(slot_count)
+
     def _execute_simulation(self, slot_resume: int, tick_resume: int,
                             console: NonBlockingConsole = None) -> None:
+        if ConstSettings.MASettings.MARKET_TYPE == SpotMarketTypeEnum.COEFFICIENTS:
+            self._execute_coefficient_simulation(slot_resume, tick_resume, console)
+        else:
+            self._execute_market_simulation(slot_resume, tick_resume, console)
```

Comment:
Nit: I would generally go with two separate classes (e.g. `CoefficientSimulation` and ` MarketSimulation`) rather than implementing two mutually-excluding functions in the same class. Is there a specific reason to go this way instead, or was it a matter of time?